### PR TITLE
[examples] Fix ImageClassification invalid probability

### LINKED ISSUE #1574: [examples] ImageClassification gives invalid probability.
## Description
As discussion #1566, `examples.inference.ImageClassification` gives probability >0 or <1.

## What have you tried to solve it?

Set `.optApplySoftmax(true)` to set a Softmax classification layer.

### PR DESCRIPTION
**Pull request by dandansamax**
## Description ##

- Set a softmax classification layer in `examples.inference.ImageClassification`.
- Write a test for Class `ImageClassification`
- close #1574

### CODE REVIEW

File: examples/src/test/java/ai/djl/examples/inference/ImageClassificationTest.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
+/*
```
Comment (frankfliu) [inline examples/src/test/java/ai/djl/examples/inference/ImageClassificationTest.java:1]: We don't need this test, we have the same test in TrainMnistTest.java

Comment (dandansamax) [reply examples/src/test/java/ai/djl/examples/inference/ImageClassificationTest.java:1]: OK, I miss that. I will delete it.

Comment (frankfliu) [reply examples/src/test/java/ai/djl/examples/inference/ImageClassificationTest.java:1]: Please add check the value should be less than 1 in TrainMnistTest.java

Comment (dandansamax) [reply examples/src/test/java/ai/djl/examples/inference/ImageClassificationTest.java]: I'm a little confused about the if statement `Boolean.getBoolean("nightly")`. When does it become true?

Comment (frankfliu) [reply examples/src/test/java/ai/djl/examples/inference/ImageClassificationTest.java]: We have continuous and nightly build, see:
https://github.com/deepjavalibrary/djl/blob/master/.github/workflows/nightly_publish.yml#L37

**Review by frankfliu — APPROVED** [review]

**Pull request merged**